test/templates/template_test.rb

```diff
@@ -17,15 +17,25 @@ def setup
 
   def test_template_requests_return_data_and_contain_template_url
     @expected_body = "my template"
+    @args = { :token => "test-token" }
     stub_request(:get, @foreman_url+'/unattended/provision?token=test-token&url='+@template_url).to_return(:status => [200, 'OK'], :body => @expected_body)
-    result = Proxy::Templates::Handler.get_template('provision', 'test-token')
+    result = Proxy::Templates::Handler.get_template('provision', @args)
     assert_equal(@expected_body, result)
   end
 
   def test_template_requests_pass_static_flag_to_foreman
     @expected_body = "my template"
+    @args = { :token => "test-token" }
     stub_request(:get, @foreman_url+'/unattended/provision?static=true&token=test-token&url='+@template_url).to_return(:status => [200, 'OK'], :body => @expected_body)
```

Comment:
Just a note, your not required to use stub_request if you use the following mixin `include Rack::Test::Methods`

With this mixin you can just use get, post, put, ... like the following example

```ruby
test_args = { :bmc_provider => 'freeipmi' }
get "/#{host}/lan/gateway", test_args
assert_equal 'unauthorized', last_response.body
assert_equal 401, last_response.status
```
Where last_response is automatically generated by the mixin.

Also note, if you want to pass a body you would do:

```ruby
get "/#{host}/lan/gateway", test_args.to_json`, "CONTENT_TYPE" => "application/json" 
```